blob/service.go

```diff
@@ -86,6 +89,47 @@ func DefaultSubmitOptions() *SubmitOptions {
 	}
 }
 
+type BlobsubResponse struct {
+	blobs  []*Blob
+	height uint64
+}
+
+func (s *Service) Subscribe(ctx context.Context, ns share.Namespace) (<-chan *BlobsubResponse, error) {
```

Comment:
These should follow GetAll semantics on the plurality of namespaces. We have yet to decide whether to keep GetAll with a single namespace or multiple; the subscription should be consistent with it, so we should decide on that first.

Reply:
(I would also call it SubscribeAll)

Reply:
I don't think we should have multiple namespaces for subscription.  For GetAll it's fine since it is a single call, but for a long-term subscription, it adds complexity in handling on the user's side. So, if you insist on consistency, I'd rather make GetAll with a single namespace than SubscribeAll with multiple namespaces.

Reply:
You would need to write two codepaths for the same blob handling logic, if these are inconsistent. As a dev, I would hate doing the same thing twice.

Reply:
On the single vs multiple namespace. Try to imagine how the implementation of the protocol that relies on multiple namespaces would look like. If you do it sequentially, you add another roundtrip. If you do it concurrently, you bring complexity for devs to synchronise both.

Reply:
For the reasons I stated in the description, I will not be extending this to add multiple namespaces. I also do not think consistency is so important here, these are two very different types of methods - and as I described, a single channel per namespace makes the most sense for multiple namespace handling anyways.

Changing GetAll is out of scope here and so I would defer that to a discussion. I don't think this feature should be blocked on this at all

Reply:
> Try to imagine how the implementation of the protocol that relies on multiple namespaces

Having multiple subscriptions and a generic logic that parses them they will directly know from which namespace the blob comes.